src/net/server/mod.rs

```diff
@@ -0,0 +1,245 @@
+use std::net::SocketAddr;
+
+use axum::{
+    extract::rejection::QueryRejection,
+    response::{IntoResponse, Response},
+    routing::get,
+    Router,
+};
+use axum_server::{tls_rustls::RustlsConfig, Handle};
+use hyper::StatusCode;
+use thiserror::Error;
+use tokio::task::JoinHandle;
+
+mod handlers;
+
+#[derive(Error, Debug)]
+pub enum MpcServerError {
+    #[error(transparent)]
+    BadQueryString(#[from] QueryRejection),
+}
+
+impl IntoResponse for MpcServerError {
+    fn into_response(self) -> Response {
+        let status_code = match &self {
+            MpcServerError::BadQueryString(_) => StatusCode::BAD_REQUEST,
+        };
+
+        (status_code, self.to_string()).into_response()
+    }
+}
+
+/// Axum router definition for MPC helper endpoint
+#[must_use]
+pub fn router() -> Router {
+    Router::new().route("/echo", get(handlers::echo_handler))
```

Comment:
so the idea is that you would add a route here eventually for MPC comms?

Reply:
that's right, however we can also create many endpoints with different routers attached to them.